common/datavalues/src/types/data_type.rs

```diff
@@ -125,7 +128,26 @@ pub trait DataType: std::fmt::Debug + Sync + Send + DynClone {
 
     fn create_mutable(&self, capacity: usize) -> Box<dyn MutableColumn>;
 
-    fn create_serializer(&self) -> TypeSerializerImpl;
+    fn create_serializer_const<'a>(&self, col: &'a ColumnRef) -> Result<TypeSerializerImpl<'a>> {
```

Comment:
Maybe we can merge `create_serializer_const` into `create_serializer`?

Reply:
oh, I got you ..., you are right, I will refator it

Reply:
done, thanks